File: files/CH-083.py
import pandas as pd

path = "CH-083 Custom splitter 3.xlsx"
input = pd.read_excel(path, usecols="B", nrows = 1, skiprows = 1)
test = pd.read_excel(path, usecols="D:F", skiprows = 4)

result = input['Info'].str.split(';', expand=True).stack().str.split(", ", expand=True)\
    .rename(columns={0: 'Date', 1: 'Product', 2: "Quantity"}).reset_index(drop=True)
result['Quantity'] = result['Quantity'].astype('int64')

print(result.equals(test)) # True

File: files/000-099/083/CH-083.R
library(tidyverse)
library(readxl)

path = "files/CH-083 Custom splitter 3.xlsx"
input = read_excel(path, range = "B2:B3")
test = read_excel(path, range = "D5:F27")

result = input %>%
  separate_rows(Info, sep = ";") %>%
  separate(Info, into = c("Date","Product","Quantity"), sep = ", ") %>%
  mutate(Quantity = as.numeric(Quantity))

identical(result, test)
#> [1] TRUE